Build fails on FreeBSD 14.2

**Describe the bug**

FreeBSD build fails with this error:

```shell
error[E0308]: mismatched types
   --> base/error.rs:123:13
    |
104 |         match self {
    |               ---- this expression has type `&Errno`
...
123 |             Error::EOPNOTSUPP => ErrorKind::Unimplemented,
    |             ^^^^^^^^^^^^^^^^^ expected `&Errno`, found `Errno`
    |
help: consider dereferencing to access the inner value using the Deref trait
    |
104 |         match &*self {
    |               ++
```

**To Reproduce**

1. Clone repository on a FreeBSD machine
2. Change to `server` directory
3. Run `cargo build --release`

**Expected behavior**

To build successfully

**Additional context**

Not sure why this is happening. Happy to pop a PR in to conditionally change the line to this if the target is FreeBSD: 

```diff
diff --git a/server/base/error.rs b/server/base/error.rs
index 9116e52..2d38f4a 100644
--- a/server/base/error.rs
+++ b/server/base/error.rs
@@ -120,6 +120,9 @@ impl ToErrKind for nix::Error {
             Error::EBADF | Error::EFAULT => ErrorKind::InvalidArgument,
             Error::EINTR | Error::EAGAIN => ErrorKind::Aborted,
             Error::ENOENT | Error::ENODEV => ErrorKind::NotFound,
+            #[cfg(target_os = "freebsd")]
+            &Error::EOPNOTSUPP => ErrorKind::Unimplemented,
+            #[cfg(not(target_os = "freebsd"))]
             Error::EOPNOTSUPP => ErrorKind::Unimplemented,
             _ => ErrorKind::Unknown,
         }
```

This does fix the issue but irks me that it's an underlying issue with `nix`.

--- a/server/base/error.rs
+++ b/server/base/error.rs
@@ -101,7 +101,7 @@ impl ToErrKind for rusqlite::types::FromSqlError {
 impl ToErrKind for nix::Error {
     fn err_kind(&self) -> ErrorKind {
         use nix::Error;
-        match self {
+        match *self {
             Error::EACCES | Error::EPERM => ErrorKind::PermissionDenied,
             Error::EDQUOT => ErrorKind::ResourceExhausted,
             Error::EBUSY

